master/internal/core.go

```diff
@@ -476,10 +479,31 @@ func (m *Master) startServers(ctx context.Context, cert *tls.Certificate) error
 
 	// If configured, set up TLS wrapping.
 	if cert != nil {
+		var clientCAs *x509.CertPool
+		clientAuthMode := tls.NoClientCert
+
+		if agentRM := m.config.ResourceManager.AgentRM; agentRM != nil && agentRM.RequireAuthentication {
+			clientAuthMode = tls.VerifyClientCertIfGiven
```

Comment:
Should we set this to `RequireAndVerifyClientCert` instead? If we request it but don't enforce that it's given, then couldn't a malicious party just bypass security by not providing a certificate?